### DISCUSSION TITLE
How to simulate IP Cores in CocoTB?

### DISCUSSION THREAD

**Discussion by tarik-hamedovic** [cocotb/cocotb#3839] [Q&A] [answered]
I have an IP Core generated by Lattice Diamond IDE and I wish to simulate the DUT in CocoTB?
But everytime I add files to the makefile and run 'make' in terminal I get some error.
It says the modules are missing: AND2,  CCU2C, FD1P3DX, INV, MUX21, ROM16X1A, ROM64X1A, VHI, VLO, XOR2
When I add other .v files in VERILOG_SOURCES I get that Module was already declared
There are lots of other files that were generated besided the .v file: .tcl, .arearep, .edn, .cst, .fdc, .ldc, .lpc, .lpf, .lsedata, .naf, .ngd, .ngo, .sort, .srp, .sym, .synproj, .twr, .asd
I have tried including them with VERILOG_INCLUDE_DIRS but nothing happens.

Is there a way to simulate IP Cores in CocoTB, if so how does the makefile look like?

**Comment by ktbarrett** [comment, marked as answer]
Some of those listed modules are unisim modules. You'll need to include the generated unisim source files. I'm not sure where your simulator places them or what flow is necessary to generate them, that's something you'll have to work out.

It might also be easier to integrate cocotb into whatever simulation flow your project management tool (such as Vivado) provides by following the directions [here](https://docs.cocotb.org/en/stable/custom_flows.html).
